torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -175,35 +198,63 @@ struct LiftedIValueArg {
 };
 
 struct LiftedIValueArgs {
+  LiftedIValueArgs(const std::optional<size_t>& active_node_call_idx)
+      : active_node_call_idx(active_node_call_idx) {}
+
   at::IValue& next_proxy(const at::IValue* actual_ptr) {
     TORCH_INTERNAL_ASSERT(next < args.size());
     auto& iv_arg = args.at(next++);
     TORCH_INTERNAL_ASSERT(iv_arg.actual_ptr == actual_ptr);
     return iv_arg.proxy;
   }
 
+  void add(const at::IValue* iv) {
+    args.emplace_back(iv);
+    if (active_node_call_idx.has_value()) {
+      args_origins.emplace_back(active_node_call_idx.value());
+    }
+  }
+
   std::vector<LiftedIValueArg> args;
   size_t next = 0;
+  // NodeCall id of each arg, only when verbose logging is enabled
+  std::vector<uint32_t> args_origins;
+
+ private:
+  const std::optional<size_t>& active_node_call_idx;
 };
 
 struct AutogradCompilerCall {
+  AutogradCompilerCall()
```

Comment:
Should this have been something like
`AutogradCompilerCall(const std::optional<size_t>& active_node_call_idx)`

Reply:
No, there should be 1 AutogradCompilerCall instance for the entire autograd graph. Is this causing an error?

Reply:
I'm seeing the following warning (not an error)
```
In file included from /home/isuruf/git/pytorch/torch/csrc/autograd/custom_function.h:10,
                 from /home/isuruf/git/pytorch/torch/csrc/autograd/python_function.h:6,
                 from /home/isuruf/git/pytorch/torch/csrc/autograd/python_autograd.h:14,
                 from /home/isuruf/git/pytorch/torch/csrc/THP.h:28,
                 from /home/isuruf/git/pytorch/torch/csrc/jit/python/pybind.h:9,
                 from /home/isuruf/git/pytorch/torch/csrc/lazy/python/init.cpp:5:
/home/isuruf/git/pytorch/torch/csrc/dynamo/compiled_autograd.h: In constructor 'torch::dynamo::autograd::AutogradCompilerCall::AutogradCompilerCall()':
/home/isuruf/git/pytorch/torch/csrc/dynamo/compiled_autograd.h:229:21: warning: member 'torch::dynamo::autograd::AutogradCompilerCall::active_node_call_idx' is used uninitialized [-Wuninitialized]
  229 |       : tensor_args(active_node_call_idx),
      |                     ^~~~~~~~~~~~~~~~~~~~
/home/isuruf/git/pytorch/torch/csrc/dynamo/compiled_autograd.h:230:28: warning: member 'torch::dynamo::autograd::AutogradCompilerCall::active_node_call_idx' is used uninitialized [-Wuninitialized]
  230 |         lifted_ivalue_args(active_node_call_idx) {}
      |                            ^~~~~~~~~~~~~~~~~~~~
```

Reply:
strange, i can't repro -Wuninitialized, but this should fix it: https://github.com/pytorch/pytorch/pull/137539